src/rgw/driver/rados/rgw_zone.cc

```diff
@@ -1230,6 +1230,20 @@ int SiteConfig::load(const DoutPrefixProvider* dpp, optional_yield y,
   return r;
 }
 
+// So we don't leak memory
+struct FakeSiteConfig : public SiteConfig {
+  std::unique_ptr<RGWZoneGroup> zonegroupstorage =
+    std::make_unique<RGWZoneGroup>();
+  std::unique_ptr<RGWZone> zonestorage = std::make_unique<RGWZone>();
+};
+
+std::unique_ptr<SiteConfig> SiteConfig::make_fake() {
+  auto fake = std::make_unique<FakeSiteConfig>();
+  fake->zonegroup = fake->zonegroupstorage.get();
+  fake->zone = fake->zonestorage.get();
+  return fake;
```

Comment:
you can use the `std::optional<RGWZoneGroup> local_zonegroup` member for this storage, you don't need to derive a FakeSiteConfig:
```c++
void SiteConfig::init_fake()
{ 
  local_zonegroup.emplace(default_zonegroup_name, default_zonegroup_name);     
  
  zone_params = RGWZoneParams{default_zone_name, default_zone_name};           
  auto &entry = local_zonegroup->zones[zone_params.id];                        
  entry.id = default_zone_name;                                                
  entry.name = default_zone_name;
  
  zone = &entry;                                                               
  zonegroup = &*local_zonegroup;
}
```
like the `SiteConfig::load_local_zonegroup()` case, the `zone` and `zonegroup` pointers point into `local_zonegroup`'s memory

Reply:
Done!